setup.py

```diff
@@ -160,30 +157,40 @@ def _with_bounds(req):
         return ''.join(with_bounds)
 
 
-REQ_PATTERN = re.compile("([^=<>]+)([<=>]{1,2})(.*)")
+REQ_PATTERN = re.compile("(?P<name>[^=<>]+)(?P<comp>[<=>]{1,2})(?P<spec>[^;]+)"
+                         "(?:(;\W*python_version\W*(?P<pycomp>[<=>]{1,2})\W*"
+                         "(?P<pyspec>[0-9\.]+)))?")
 
 
-def _conda_format(req, selector=None):
-    match = REQ_PATTERN.match(req)
-    if match and match.group(1).lower() == 'numpy':
-        line = 'numpy x.x'
-    else:
-        line = REQ_PATTERN.sub(
-            lambda m: '%s %s%s' % (m.group(1).lower(), m.group(2), m.group(3)),
-            req,
-            1,
-        )
+def _conda_format(req):
+    def _sub(m):
+        name = m.group('name').lower()
+        if name == 'numpy':
+            return 'numpy x.x'
+
+        formatted = '%s %s%s' % ((name,) + m.group('comp', 'spec'))
+        pycomp, pyspec = m.group('pycomp', 'pyspec')
+        if pyspec:
+            comp = STR_TO_CMP[pycomp]
+            pyver_spec = StrictVersion(pyspec)
+            selector = []
+
+            for pyver in ('2.7', '3.4'):
```

Comment:
I think we could just do something like: `selector = 'int(py) %s %s' % (pycomp, pyver_spec)`. Conda selectors have the name `py` available as a 2 character string like `'34'`.